plugins/askrene/mcf.c

```diff
@@ -541,16 +540,13 @@ static void linear_network_add_adjacenct_arc(
 		const u32 node_idx,
 		const struct arc arc)
 {
-	assert(arc.idx < tal_count(linear_network->arc_tail_node));
+	assert(arc.idx < linear_network->max_num_arcs);
 	linear_network->arc_tail_node[arc.idx] = node_idx;
 
-	assert(node_idx < tal_count(linear_network->node_adjacency_first_arc));
+	assert(node_idx < linear_network->max_num_arcs);
```

Comment:
```suggestion
	assert(node_idx < linear_network->max_num_nodes);
```

Reply:
If the name begins with `node_` it indicates a node property, an array as big as the number of nodes.

Reply:
Doh!  It's even called "node_idx"!